avsomr/avslog.h

```diff
@@ -0,0 +1,24 @@
+#ifndef AVS_LOG_H
+#define AVS_LOG_H
+
+#include <QDebug>
+
+#define LOGD() qDebug()
+#define LOGI() qInfo()
+#define LOGW() qWarning()
+#define LOGE() qCritical()
+
+#define IF_ASSERT(cond) \
+if(!(cond)) { \
+    LOGE() << "\"ASSERT FAILED!\":" << #cond << __FILE__ << __LINE__; \
+    Q_ASSERT(cond); \
+} \
+if(!(cond)) \
```

Comment:
dito. Also beware of side effects, from evaluating `cond` twice!

Reply:
I hope that no one will do such `IF_ASSERT(++counter == 0)`, as well as it can not be done in Q_ASSERT